File: leetcode/search/441. Arranging Coins/js/main.js
// Time O(N)
// Space O(1)
const arrangeCoins = n => {
  let i = 1;
  let cnt = 0;

  while (true) {
    n = n - i;
    i++;

    if (n < 0) {
      return cnt;
    }

    cnt++;
  }
};

File: leetcode/search/441. Arranging Coins/py/main.py
# Time O(LogN)
# Space O(1)
class Solution:
    def arrangeCoins(self, totalCoins: int) -> int:
        if totalCoins <= 1:
            return totalCoins
    
        lo, hi = 0, totalCoins

        while lo <= hi:
          steps = lo + ((hi - lo) >> 1)
          coins = steps * (steps + 1) // 2
          
          if coins == totalCoins:
            return steps
            
          if coins < totalCoins:
            lo = steps + 1
          else:
            hi = steps - 1
            
        return hi
